src/main/java/redis/clients/jedis/JedisPool.java

```diff
@@ -333,21 +330,20 @@ public Jedis getResource() {
   }
 
   @Override
-  protected void returnBrokenResource(final Jedis resource) {
+  public void returnBrokenResource(final Jedis resource) {
     if (resource != null) {
       returnBrokenResourceObject(resource);
     }
   }
 
   @Override
-  protected void returnResource(final Jedis resource) {
+  public void returnResource(final Jedis resource) {
     if (resource != null) {
       try {
         resource.resetState();
         returnResourceObject(resource);
       } catch (Exception e) {
         returnBrokenResource(resource);
-        throw new JedisException("Resource is returned to the pool as broken", e);
```

Comment:
I agree not to throw a `JedisException`.
but we should logger.error for it (normal resource became to broken)

Reply:
acknowledged